builtin/logical/aws/iam_policies.go

```diff
@@ -0,0 +1,102 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"net/url"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+// PolicyDocument represents an IAM policy document
+type PolicyDocument struct {
+	Version   string
+	Statement []StatementEntry
+}
+
+// StatementEntry represents a statement in an IAM policy document
+type StatementEntry interface{}
```

Comment:
Would it be possible to strip this and just call `Statements` an `[]interface{}`? Just because then there'll be less objects in this package.

Reply:
So I started changing this, but then realized that IAM policy documents can have the Statement portion as either a list or a single element, so I ended up adding a custom json unmarshal function to handle that. So I wasn't able to remove this object entirely.

https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_statement.html

Reply:
Ah! Cool cool.